src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -44,13 +44,13 @@ internal object CaptureDisposable(object service)
         {
             Debug.Assert(!_disposed);
 
-            _captureDisposableCallback?.Invoke(service);
```

Comment:
I moved this invoke below the check for IDisposable, because this callback is only used at test time, and my test case was expecting it to be invoked 3 times (1 for each IDisposable service) but it was being invoked 4 times (once for the IEnumerable itself which doesn't implement IDisposable). It seemed clearer to adjust this, than to change my test to assert that onDispose should be called 4 times when there are only 3 disposable services.

Reply:
Although I wasn't sure if perhaps I discovered a bug - could you clarify, should CaptureDisposable() be called for the IEnumerable itself because that is what is happening.

Reply:
Can you add a test that explicitly adds an IEnumerable\<T\> service to the container that is disposable? 

I wouldn’t make tweaks in a PR that was trying to increase coverage unless the behavior was truly understood. That’s a way to introduce subtle behavior changes and I’d prefer if you reverted this

Reply:
@dazinator - can you address this feedback?